src/experiments.js

```diff
@@ -181,6 +138,44 @@ export function toggleExperiment(win, experimentId, opt_on,
   return on;
 }
 
+/**
+ * Calculate whether the experiment is on or off based off of the
+ * cookieFlag or the global config frequency given.
+ * @param {!Window} win
+ * @return {!Object<string, boolean>}
+ */
+function experimentToggles(win) {
+  if (toggles_) {
+    return toggles_;
+  }
+  toggles_ = Object.create(null);
+
+  // Read the default config of this build.
+  if (win.AMP_CONFIG) {
+    for (const experimentId in win.AMP_CONFIG) {
+      if (win.AMP_CONFIG.hasOwnProperty(experimentId)) {
+        const frequency = win.AMP_CONFIG[experimentId];
+        if (typeof frequency === 'number' && frequency >= 0 && frequency <= 1) {
+          toggles_[experimentId] = Math.random() < frequency;
+        }
+      }
+    }
+  }
+
+  // Read config from cookie, which can override the default config.
+  const cookieFlags = getExperimentIds(win);
+  for (let i = 0; i < cookieFlags.length; i++) {
+    let experimentId = cookieFlags[i];
+    let enabled = true;
+    if (experimentId[0] == '-') {
+      // It is a  disabling flag.
+      experimentId = experimentId.substr(1);
```

Comment:
In fact, currently there is a bug that can be reproduced by:
```
AMP.toggleExperiment('-some-experiment', true);
AMP.toggleExperiment('some-experiment', true);
AMP.toggleExperiment('some-experiment', true);
```
This will give you duplicated entries in the cookie 

That's to say, we should probably not have both "some-experiment" and "-some-experiment" in the cookie.